akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -359,6 +358,7 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
 
   def stopWith(closeInfo: CloseInformation, shouldAbort: Boolean = false): Unit = {
     closedMessage = Some(closeInfo)
+    unsignDeathPact()
```

Comment:
So, the watched actor would stop somewhere between here and that the unregistration completing?

Reply:
That's my guess. The `unregistered` state is so simple that I couldn't come up with any other reason why the actor should be stopped in the meantime.